pkg/database/extensions.go

```diff
@@ -0,0 +1,120 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"encoding/base64"
+
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/encryption"
+	encrypt "github.com/Azure/ARO-RP/pkg/util/encryption"
+)
+
+var _ codec.InterfaceExt = (*SecureBytesExt)(nil)
+
+type SecureBytesExt struct {
+	Cipher encryption.Cipher
+}
+
+func (s SecureBytesExt) ConvertExt(v interface{}) interface{} {
```

Comment:
this will return nil if Cipher is nil.  Is that intended?  Don't you want to return []byte(data) in that case?  Tests need to cover the overall behaviour with Cipher enabled and not.